pull/github.go

```diff
@@ -706,7 +706,12 @@ func (ghc *GitHubContext) loadPushedAt(commits []*Commit) error {
 	}
 
 	if len(commitsBySHA) > 0 {
-		return errors.Errorf("%d commits were not found while loading pushed dates", len(commitsBySHA))
+		missingSHAs := make([]string, 0, len(commitsBySHA))
+		for sha := range commitsBySHA {
+			missingSHAs = append(missingSHAs, sha)
+		}
+		return errors.Errorf("%d commits were not found while loading pushed dates. Missing %s. This may be temporary. Wait 30 seconds, and refresh this page to re-evaluate.",
```

Comment:
style:
```suggestion
		return errors.Errorf("%d commits were not found while loading pushed dates. Missing %s. This may be temporary. Wait 30 seconds then refresh this page to re-evaluate.",
```

I _think_ this will only appear on the details page (and in logs), so it's probably fine to include instructions like this here, but I think in an ideal implementation, this would mark the error as temporary in some way and only the frontend rendering code would add instructions about what to do for temporary errors.

This is an example of where having errors implement an interface or be a specific type could be helpful. On the other hand, retrying is usually a good strategy, so we could probably get 80% of the value by just adding a line in the UI that always suggests retrying on error.

Reply:
Ok, moved this to use a `TemporaryError`. I just added the troubleshooting note to the error message, but we could also add a separate UI element if you think that might be better. Perhaps a `data.ErrorTroubleshooting` passed to render?